### PR TITLE
Various fixes to lua_api.txt

### LINKED ISSUE #6079: get_inventory returns nil for Lua entities
If you register an entity with Lua using `minetest.register_entity`, the function `get_inventory` always returns `nil`. Note: The documentation claims that `get_inventory` always returns an `InvRef`. This is clearly false.
Tested in 0.4.16.

Test code:
```
minetest.register_entity("test_entity_inventory:entity", {
	visual = "cube",
	infotext = "Inventory test entity. Rightclick to show inventory.",
	on_rightclick = function(self, clicker)
		-- Get inventory
		minetest.chat_send_player(clicker:get_player_name(), "Inventory contents: " .. dump(self.object:get_inventory()))
	end,
})
```

Put this into `init.lua` in a mod named `test_entity_inventory`. Spawn the entity with `/spawnentity test_entity_inventory:entity`. A cube appears. Rightclick the cube to see the return value of `get_inventory` (see code above). I expect to get some userdata or a table, but instead I only get `nil`.

I conclude that this bug means it is currently not possible for a Lua entity to have an inventory. :disappointed:

### LINKED ISSUE #6678: Incorrect documentation for minetest.register_on_item_eat
In contrast to what the documentation says on line [2611](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L2611), returning true in the function being registered will cause an error to occur saying that it was "Expecting itemstack, itemstring, table or nil." Also, returning itemstack (the original value) causes eating to do literally nothing (including removing the item). Needless to say, it needs to be corrected.

### LINKED ISSUE #7660: LBM documentation is confusing
Actually, only LBM definition is documented, but not what LBM itself is. And it is a bit confusing:
```lua
        run_at_every_load = false,
        -- Whether to run the LBM's action every time a block gets loaded,
        -- and not just for blocks that were saved last time before LBMs were
        -- introduced to the world.
```
1. “LBMs were” should probably be “the LBM was.”
2. If an LBM is removed and then re-added, is it considered newly introduced? (According to [one comment](https://github.com/minetest-mods/technic/pull/437#discussion_r211086167), it is. But that doesn’t seem to be the intended behavior)

### LINKED ISSUE #8395: 'move_to': explain 'interpolated', document 'continuous'
'continuous' isn't even documented.
https://github.com/minetest/minetest/blob/7f1c2b8a00ede0659746102762aeb1da487eb6f8/doc/lua_api.txt#L5149
Relevant code:
https://github.com/minetest/minetest/blob/38f6e7a1981a8dcba9bfa9aa32ca77cd19a960aa/src/script/lua_api/l_object.cpp#L148
https://github.com/minetest/minetest/blob/88c68ce8ec79bd5ce60d11fb4a09df1adc720d2f/src/content_sao.cpp#L693
https://github.com/minetest/minetest/blob/88c68ce8ec79bd5ce60d11fb4a09df1adc720d2f/src/content_sao.cpp#L1204
https://github.com/minetest/minetest/blob/88c68ce8ec79bd5ce60d11fb4a09df1adc720d2f/src/content_sao.cpp#L786

### PR DESCRIPTION
**Pull request by DS-Minetest**
- Goal of the PR: Fix missing and wrong documentation. The goal is not to improve anything in the style.
- How does the PR work? It fixes documentation.
- Related PR: #7599 (Introduced wrongness with `on_punch` and `can_dig`.)
Fixes #8395. (Note: I might be wrong)
Fixes #7660.
Fixes #6678.
Fixes #6079.

## To do

This PR is a Ready for Review.

## How to test

Read it, check whether everything is correct and comment if something is wrong or can be expressed in a better way.

### CODE REVIEW

**Comment by SmallJoker** [comment]
Isn't there more to fix in the Lua API? Don't get me wrong - the change is good, but single commits like this are quite inefficient.

**Comment by DS-Minetest** [comment]
If you tell me about other wrong documentation in `lua_api.txt` that can easily be fixed without causing a controversial discussion, I'll be glad to include fixes for that in this PR, too. :slightly_smiling_face:

**Comment by SmallJoker** [comment]
#8395 (uses prediction parameter that may overshoot at the destination)
#7660 
#6678 partially
#5854 pick any
#6079

**Comment by DS-Minetest** [comment]
Ok, updated to include fixes to those issues.

#8395 was difficult. I've used the testing code posted here: https://github.com/minetest/minetest/issues/8395#issuecomment-526856660 What I wrote about `move_to` might be wrong.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -5400,7 +5399,14 @@ This is basically a reference to a C++ `ServerActiveObject`
 
 * `get_pos()`: returns `{x=num, y=num, z=num}`
 * `set_pos(pos)`: `pos`=`{x=num, y=num, z=num}`
-* `move_to(pos, continuous=false)`: interpolated move
+* `move_to(pos, continuous=false)`
+    * Does an interpolated move for lua entities. The object will not instantly
```
Comment (SmallJoker) [inline doc/lua_api.txt]: Suggestion for both sentences:
`Does an interpolated move for Lua entities for visually smooth transitions.`

File: doc/lua_api.txt [hunk 2]
```diff
@@ -5400,7 +5399,14 @@ This is basically a reference to a C++ `ServerActiveObject`
...
+      be at `pos` but can be watched how it gets there.
+    * If the objectref is a lua entity and `continuous` is true, object will not
+      be moved to the current position before starting the interpolated move.
+    * For players this is the same as `get_pos` with the only difference
+      that the mapblock in which the player ends up is not force-sent.
```
Comment (SmallJoker) [inline doc/lua_api.txt]: Force sending is an internal gameplay optimization, and irrelevant to modders or players. Also refer to `ObjectRef` (CamelCase, including code formatting) above, please.

**Comment by SmallJoker** [comment]
Very nice, thank you.

File: doc/lua_api.txt (hunk 1 above)
Comment (DS-Minetest) [reply doc/lua_api.txt]: Thanks! This is much better.

File: doc/lua_api.txt (hunk 2 above)
Comment (DS-Minetest) [reply doc/lua_api.txt]: I wonder if it was good to document the automatic block sending for `set_pos` at `send_mapblock`. :thinking: 
https://github.com/minetest/minetest/blob/6b489b64616b0f34c1a36474fdfe3feb90ac6800/doc/lua_api.txt#L5651-L5655

**Pull request merged**